core/src/main/java/org/infinispan/interceptors/ClusteringInterceptor.java

```diff
@@ -76,17 +76,17 @@ protected boolean isNeedReliableReturnValues(FlagAffectedCommand command) {
    }
 
    protected boolean needsRemoteGet(InvocationContext ctx, AbstractDataCommand command) {
-      if (command.hasFlag(Flag.CACHE_MODE_LOCAL)
-            || command.hasFlag(Flag.SKIP_REMOTE_LOOKUP)
-            || command.hasFlag(Flag.IGNORE_RETURN_VALUES)) {
+         if(command.hasFlag(Flag.CACHE_MODE_LOCAL)
+            || (command.hasFlag(Flag.SKIP_REMOTE_LOOKUP) && !command.hasFlag(Flag.DELTA_WRITE)
+            || command.hasFlag(Flag.IGNORE_RETURN_VALUES))) {
```

Comment:
Here I was suggesting to remove the `DELTA_WRITE` check only for `CACHE_MODE_LOCAL` and keep it for the other two... I'll add another commit to do that.